Standards/scs-0210-v2-k8s-version-policy.md

```diff
@@ -0,0 +1,81 @@
+Kubernetes supports its releases for around 14 months. 12 of these are the standard
+support period. The remaining 2 months are end-of-life support for things like:
+
+- CVEs (under the advisement of the Security Response Committee)
+- dependency issues (including base image updates)
+- critical core component issues
+
+More information can be found under [Kubernetes Support Period].
+
+The Kubernetes releases are planned for about 4 months, which usually results in about
+**3 minor** releases per year [Kubernetes Release Cycle].
+
+Patches to these releases are provided monthly, with the exception of the first patch,
+which is usually provided 1-2 weeks after the initial release [Patch Release Cadence].
+
+## Motivation
+
+Kubernetes is a living, fast-paced project, which follows a pre-defined release cycle.
+This enables forward planning with regards to releases and patches, but also implies a
+necessity to upgrade to newer versions quickly, since these often include new features
+or important security updates, especially if an old version falls out of the support
+period window.
+
+We want to achieve an up-to-date policy, which means that providers should be mostly in
+sync with the upstream and don't fall behind the official Kubernetes releases.
+This is achievable, since new versions are released periodical on a well communicated
+schedule. This should enable providers and users to set up processes around this.
+Being up to date ensures that security issues and bugs are addressed and new features
+are made available when using SCS compliant clusters.
+
+It is nevertheless important to at least support all Kubernetes versions that are still
+inside the support period, since users could depend on specific versions or may need
+longer to upgrade their workloads to a newer version.
+
+It is therefore reasonable to recommend a version recency policy as well as a support
+window period.
+
+## Decision
+
+In order to keep up-to date with the latest Kubernetes features, bug fixes and security improvements,
+the provided Kubernetes versions should be kept up to date with the upstream.
+
+- The latest minor version SHOULD be provided no later than 4 months after release.
+- The latest patch version SHOULD be provided no later than 1 week after release.
+- This time period MUST be even shorter for patches that target critical CVEs (CVSS >= 8).
+  It is RECOMMENDED to provide the version in a 2 day time period after release.
+- New versions SHOULD be tested before being rolled out on productive infrastructure.
+
+At the same time, provider should support Kubernetes versions at least as long as the official sources as mentioned
+in the Kubernetes Support Period document.
+
+- Kubernetes versions SHOULD be supported as long as the official sources support them.
+  At the moment, this support period is 14 months.
+- It is RECOMMENDED to not support versions after this period in order to not encourage
+  usage of out-of-date versions.
```

Comment:
As line 57 allows to delay latest minor version for 4 month, I would take this into account with respect to shut of oldest minor version. I can image, that customers may not adapt their applications to latest minor version, but may skip ones. Hence, I would rephrase this sentence in order to not remove older verisons, before new versions will be supported.


```suggestion
- It is RECOMMENDED to not support versions after this period in order to not encourage
  usage of out-of-date versions, but not before the latest minor version is supported.
```

Reply:
If I do the math:
1.N is released and provided in month 0, extended support till month 14
1.N+1 is released in month 4 and provided in month 8, extended support till month 18
1.N+2 is released in month 8 and provided in month 12, extended support till month 22
So, skipping one level from 1.N to 1.N+2 is still possible with a 2months overlap window.
Sidenote: Technically, a rolling upgrade will need to do the intermediate step (but of course that could be very short-lived).
I would not complicate the policy, as skipping one version does not require it.

Reply:
I would also not add this clause, since it complicates things as @garloff said, and the versions are provided in regular time intervals. Also we can't control what customers want to do with their applications; as long as the providers give them access to the latest versions (as the standard says), the can decide themselves what to do.